framer/GestureInputRecognizer.coffee

```diff
@@ -0,0 +1,448 @@
+
+		# Trigger a force touch if we reach the desktop threshold
+		if event.webkitForce >= GestureInputForceTapDesktop
+			@forcetapstart(event)
+		else
+			@forcetapend(event)
+
+	forcetapchange: (event) =>
+		@_dispatchEvent("forcetapchange", event)
+
+	forcetapstart: (event) =>
+		return unless @session
+		return if @session.started.forcetap
+		@session.started.forcetap = event
+		@_dispatchEvent("forcetapstart", event)
+		@_dispatchEvent("forcetap", event)
+
+	forcetapend: (event) =>
+		return unless @session
+		return unless @session.started.forcetap
+		@session.started.forcetap = null
+		@_dispatchEvent("forcetapend", event)
+
+	# Pan
+
+	panstart: (event) =>
+		@session.started.pan = event
+		@_dispatchEvent("panstart", event, @session.started.pan.target)
+
+	pan: (event) =>
+		@_dispatchEvent("pan", event, @session.started.pan.target)
+		direction = @_getDirection(event.delta)
+		@["pan#{direction}"](event) if direction
+		
+	panend: (event) =>
+		@_dispatchEvent("panend", event, @session.started.pan.target)
+		@session.started.pan = null
+
+	panup: (event) => @_dispatchEvent("panup", event, @session.started.pan.target)
+	pandown: (event) => @_dispatchEvent("pandown", event, @session.started.pan.target)
+	panleft: (event) => @_dispatchEvent("panleft", event, @session.started.pan.target)
+	panright: (event) => @_dispatchEvent("panright", event, @session.started.pan.target)
+
+	# Pinch
+
+	pinchstart: (event) =>
+		@session.started.pinch = event
+		@scalestart(event, @session.started.pinch.target)
+		@rotatestart(event, @session.started.pinch.target)
+		@_dispatchEvent("pinchstart", event)
+
+	pinch: (event) =>
+		@_dispatchEvent("pinch", event)
+		@scale(event, @session.started.pinch.target)
+		@rotate(event, @session.started.pinch.target)
+		
+	pinchend: (event) =>
+		@_dispatchEvent("pinchend", event)
+		@scaleend(event, @session.started.pinch.target)
+		@rotateend(event, @session.started.pinch.target)
+		@session.started.pinch = null
+
+
+	scalestart: (event) => @_dispatchEvent("scalestart", event)
+	scale: (event) => @_dispatchEvent("scale", event)
+	scaleend: (event) => @_dispatchEvent("scaleend", event)
+
+	rotatestart: (event) => @_dispatchEvent("rotatestart", event)
+	rotate: (event) => @_dispatchEvent("rotate", event)
+	rotateend: (event) => @_dispatchEvent("rotateend", event)
+
+	# Swipe
+	
+	swipestart: (event) =>
+		@_dispatchEvent("swipestart", event)
+		@session.started.swipe = event
+		@swipedirectionstart(event)
+
+	swipe: (event) =>
+		@_dispatchEvent("swipe", event)
+		@swipedirection(event)
+
+	swipeend: (event) =>
+		@_dispatchEvent("swipeend", event)
+
+	# Direction swipe
+
+	swipedirectionstart: (event) =>
+		return unless event.offsetDirection
+		return if @session.started.swipedirection 
+		@session.started.swipedirection = event
+		direction = @session.started.swipedirection.offsetDirection
+		@_dispatchEvent("swipe#{direction}start", event)
+
+		swipeEdge = @_edgeForSwipeDirection(direction)
```

Comment:
If this is what I think it is, that's **awesome** :+1: :+1: 

Btw I am not able to trigger any 'edge' events using the Beta on the laptop, but maybe I'm doing it wrong.